src/snitch_registry.cpp

```diff
@@ -480,7 +492,8 @@ impl::test_state registry::run(impl::test_case& test) noexcept {
     impl::test_state state{
         .reg = *this, .test = test, .may_fail = may_fail, .should_fail = should_fail};
 
-    state.locations.push_back(test.location);
+    state.locations.push_back(
```

Comment:
nit: maybe better to use `emplace_back` here to construct the object directly

Reply:
`snitch::small_vector` doesn't have `emplace_back()`. Even if we add this missing function, for it to be useful we'd have to refactor the vector implementations, since it currently always default-initialize elements when growing (rather than, like `std::vector`, just grow the memory buffer with uninitialized data). 

I had made this conscious choice so the class could be fully `constexpr`, and also to keep the code simple so it compiles fast (though I never benchmarked that!). But we could revisit it at some point; this could improve performance across the board.